linalg/hypre.cpp

```diff
@@ -31,6 +31,47 @@ using namespace std;
 namespace mfem
 {
 
+
+HYPRE_Session::HYPRE_Session()
+{
+#if MFEM_HYPRE_VERSION >= 21500
+   HYPRE_Init();
+#endif
+
+   SetGlobalOptions();
+}
+
+HYPRE_Session::~HYPRE_Session()
+{
+#if MFEM_HYPRE_VERSION >= 21500
+   HYPRE_Finalize();
+#endif
+}
+
+void HYPRE_Session::SetGlobalOptions()
+{
+#if MFEM_HYPRE_VERSION >= 22100
+   // Use hypre's SpGEMM instead of cuSPARSE.
+   HYPRE_SetSpGemmUseCusparse(0);
+#endif
+
+#if 0
+   // Additional global options, see
+   // https://hypre.readthedocs.io/en/latest/solvers-boomeramg.html#gpu-supported-options
+   // for more details
+
+   // AMG in GPU memory (default)
+   HYPRE_SetMemoryLocation(HYPRE_MEMORY_DEVICE);
+   // Setup AMG on GPUs
+   HYPRE_SetExecutionPolicy(HYPRE_EXEC_DEVICE);
+   // Use GPU RNG
+   HYPRE_SetUseGpuRand(1);
+   // Umpire memory pool
+   HYPRE_SetUmpireUMPoolName("HYPRE_UM_POOL_TEST");
+   HYPRE_SetUmpireDevicePoolName("HYPRE_DEVICE_POOL_TEST");
+#endif
```

Comment:
If those functions can be called multiple times, I suggest we setup a sane default like mentioned in the hypre documentation. Don't leave the `#if 0` here.

Reply:
Most definitely 👍

Reply:
Resolved now?